lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -4,378 +4,95 @@
-    protected $keepSlave = false;
-
-    /**
-     * Creates Master Slave Connection.
+     * Creates Primary Replica Connection.
      *
      * @param mixed[] $params
      *
      * @throws InvalidArgumentException
      */
     public function __construct(array $params, Driver $driver, ?Configuration $config = null, ?EventManager $eventManager = null)
     {
-        if (! isset($params['slaves'], $params['master'])) {
-            throw new InvalidArgumentException('master or slaves configuration missing');
-        }
-
-        if (count($params['slaves']) === 0) {
-            throw new InvalidArgumentException('You have to configure at least one slaves.');
-        }
-
-        $params['master']['driver'] = $params['driver'];
-        foreach ($params['slaves'] as $slaveKey => $slave) {
-            $params['slaves'][$slaveKey]['driver'] = $params['driver'];
-        }
-
-        $this->keepSlave = (bool) ($params['keepSlave'] ?? false);
-
-        parent::__construct($params, $driver, $config, $eventManager);
-    }
-
-    /**
-     * Checks if the connection is currently towards the master or not.
-     *
-     * @return bool
-     */
-    public function isConnectedToMaster()
-    {
-        return $this->_conn !== null && $this->_conn === $this->connections['master'];
-    }
-
-    /**
-     * @param string|null $connectionName
-     *
-     * @return bool
-     */
-    public function connect($connectionName = null)
-    {
-        $requestedConnectionChange = ($connectionName !== null);
-        $connectionName            = $connectionName ?: 'slave';
-
-        if ($connectionName !== 'slave' && $connectionName !== 'master') {
-            throw new InvalidArgumentException('Invalid option to connect(), only master or slave allowed.');
-        }
+        $this->deprecated(self::class, PrimaryReplicaConnection::class);
 
-        // If we have a connection open, and this is not an explicit connection
-        // change request, then abort right here, because we are already done.
-        // This prevents writes to the slave in case of "keepSlave" option enabled.
-        if ($this->_conn !== null && ! $requestedConnectionChange) {
-            return false;
-        }
-
-        $forceMasterAsSlave = false;
+        if (isset($params['master'])) {
+            $this->deprecated("Params key 'master'", "'primary'");
 
-        if ($this->getTransactionNestingLevel() > 0) {
-            $connectionName     = 'master';
-            $forceMasterAsSlave = true;
+            $params['primary'] = $params['master'];
+            unset($params['master']);
         }
 
-        if (isset($this->connections[$connectionName])) {
-            $this->_conn = $this->connections[$connectionName];
-
-            if ($forceMasterAsSlave && ! $this->keepSlave) {
-                $this->connections['slave'] = $this->_conn;
-            }
+        if (isset($params['slaves'])) {
+            $this->deprecated("Params key 'slaves'", "'replica'");
```

Comment:
I'd rather use quotes than apostrophs around the key names in the message (I know that devs think about `'` being a single quote, but that's not what it is in English sentences).